sql/analyzer/analyzer.go

```diff
@@ -521,7 +521,44 @@ func (a *Analyzer) analyzeWithSelector(ctx *sql.Context, n sql.Node, scope *plan
 	)
 	a.Log("starting analysis of node of type: %T", n)
 	a.LogNode(n)
-	for _, batch := range a.Batches {
+
+	batches := a.Batches
+	switch n := n.(type) {
+	case *plan.Commit:
+		batches = nil
+	case *plan.StartTransaction:
+		batches = nil
+	case *plan.InsertInto:
+		if n.SkipSourceAnalyze {
+			batches = []*Batch{
```

Comment:
This seems kind of fraught, prone to getting forgotten as rules are added and changed. It would be best to express this list in terms of a mutation to another batch. Failing that, define it next to the main list and reference it here.

Reply:
Yeah, bit of a tradeoff of (1) hard to break or (2) hard to understand. The case being selected here is basically just `INSERT ... VALUES (), (),...`, and probably simple UPDATE/DELETE of the same shape soon -- these seem pretty hard to miss in testing?